multinet/api.py

```diff
@@ -22,76 +31,168 @@ def require_db():
 bp.before_request(require_db)
 
 
-def graphql_query(query, variables=None):
-    """Perform a GraphQL query using optional variable definitions."""
-    data = None
-    errors = []
+@bp.route("/workspaces", methods=["GET"])
+def get_workspaces():
+    """Retrieve list of workspaces."""
+    return json.dumps(db.get_workspaces())
 
-    result = graphql(schema, query, variables=variables or {})
-    if result:
-        errors = [error.message for error in result.errors] if result.errors else []
-        data = result.data
 
-        if errors:
-            app.logger.error("Errors in request: %s" % len(errors))
-            for error in errors[:10]:
-                app.logger.error(error)
+def lookup_workspace(workspace):
+    """Retrieve a single workspace, with a status code."""
+    result = db.get_workspace(workspace)
+    return (workspace, 404 if result is None else 200)
```

Comment:
It seems here we're just returning the name of the workspace, and not the result itself. I think here we should do something like:
```
result = db.get_workspace(workspace) or None
status = 200 if result else 404
return (result, status)
```
@waxlamp Lmk what you think

Reply:
If `result` is `None` then you still want to return the name of the workspace as the response, just with the 404 code indicating which workspace was Not Found. The target use case here is if you sent out a bunch of requests in parallel for workspaces, then when they resolve eventually you'll want to make sure you know which one a 404 code is talking about.